transport/src/commonMain/kotlin/com/genesys/cloud/messenger/transport/core/MessagingClientImpl.kt

```diff
@@ -52,6 +54,7 @@ internal class MessagingClientImpl(
     private val eventHandler: EventHandler = EventHandlerImpl(log.withTag(LogTag.EVENT_HANDLER)),
     private val userTypingProvider: UserTypingProvider = UserTypingProvider(log.withTag(LogTag.TYPING_INDICATOR_PROVIDER)),
     private val healthCheckProvider: HealthCheckProvider = HealthCheckProvider(log.withTag(LogTag.HEALTH_CHECK_PROVIDER)),
+    private var deploymentConfig: KProperty0<DeploymentConfig?>,
```

Comment:
I admit, I'm not familiar with this construct and reading the comment on it doesn't help me get any further. Can you explain the benefit/purpose? I note the config isn't used yet so it's hard to tell how it's accessed (presumedly `get()`) and what it buys us.

Reply:
This is just a reference to the MessengerTransport.depoloymentConfig property getter (read only).  
In case of `Conversation Autostart` we are going to read the `deploymentConfig.messenger.apps.conversations.autoStart.enabled` field when we receive the SessionResponse to check the autostart configuration and if it is enabled we are going to execute the AutostartRequest.

Reply:
I've never seen `KProperty0`. How is that different from just `DeploymentConfig?`?

Reply:
If you pass the `deploymentConfig` in MessagingClient constructor you will get a copy of this value. And when it will be updated in MessengerTransport you will not receive the update. When you pass the KProperty0 in the constructor it is the same as passing a reference to the deploymentConfig.get() function. So every time you call it you call the deploymentConfig getter.

Reply:
interface KProperty0<out R>
"Represents a property without any kind of receiver. Such property is either originally declared in a receiverless context such as a package, or has the receiver bound to it."
😵‍💫